core/services/ardupilot_manager/ArduPilotManager.py

```diff
@@ -104,3 +116,31 @@ def start_board(self, boards: List[Tuple[FlightControllerType, str]]) -> bool:
 
     def restart(self) -> None:
         self.mavlink_manager.restart()
+
+    def _load_endpoints(self) -> None:
+        if "endpoints" not in self.configuration:
+            self.configuration["endpoints"] = []
+        else:
+            endpoints = self.configuration["endpoints"]
+            self.mavlink_manager.add_endpoints([Endpoint(e) for e in endpoints])
+
+    def get_endpoints(self) -> List[Endpoint]:
+        return self.mavlink_manager.endpoints()
+
+    def add_endpoint(self, endpoint: Endpoint) -> bool:
+        if self.mavlink_manager.add_endpoint(endpoint):
```

Comment:
Just a tip, you can avoid the code in the extra indentation layer using the invert return logic.